docs/04-contributing/docker.md

```diff
@@ -0,0 +1,153 @@
+
+There's minimal overlap between most services and their `Dockerfile`s. Here are a few things to know.
+
+### Split compose file
+
+Instead of having one large compose file to cater to every need, there are multiple - the main one being `compose.yml`.
+This takes advantage of [merging][docker compose merging]. Merging is done automatically when a `compose.override.yml`
+is present, otherwise the list of files has to be passed with `docker compose -f compose.yml -f file1.yml -f file2.yml`.
+
+### php/Dockerfile
+
+[This][php-dockerfile] is the biggest and most complex `Dockerfile` in the repo.
+It's a multi-stage file that attempts to make the final image minimal. You can read it, but the most important things
+to keep in mind are:
+
+- the `base` target only has the minimal common items for the next steps
+- for production there are builder targets to build the frontend and backend of mbin
+- for development, the `dev` target is minimal as the frontend and backend are built when the services are first run 
+- the final target is `prod` which unites the outputs of the builder targets
+
+### Building the frontend and backend
+
+As mentioned [above](#phpdockerfile), the front and backend are built in the `Dockerfile` for production.
+
+In development, this is taken care of at runtime in [compose.dev.yml]. The `messenger` service is the first PHP
+service to run, so it installs PHP dependencies and such. The `node` service is only in the dev compose not production
+and requires files from the PHP dependencies to successfully build the frontend. It is thus run afterward.
+
+
+## Frequently executed tasks
+
+### Modifying behavior with a quick turn-around
+
+The entire repository is mounted into the `caddy`, `php`, and `messenger` services.
+
+**PHP code**
+
+Most PHP code changes are picked up immediately after refreshing the page in the browser.
+
+**Other important files**
+
+| file          | services       | how to refresh                                                         |
+|---------------|----------------|------------------------------------------------------------------------|
+| Caddyfile     | caddy          | `docker compose exec caddy caddy reload --config /etc/caddy/Caddyfile` |
+| entrypoint.sh | php, messenger | `docker compose up -d --no-deps $service`                              |
+
+
+### Debugging PHP (XDebug)
+
+PHP supports remote debugging using [XDebug](https://xdebug.org/). This works by hosting an Xdebug server the php
+process can call. 
+
+```mermaid
+sequenceDiagram
+    participant Browser
+    participant rp as Reverse Proxy
+    participant php as PHP-FPM
+    participant xdebug as XDebug Host
+
+    Browser->>+rp: http://localhost
+    rp->>+php: index.php
+    php->>+xdebug: :9003
+    xdebug->>php: break
+    php-->>xdebug: 
+    xdebug->>php: continue
+    php-->>xdebug: 
+    xdebug-->>-php: done
+    php-->>-rp: response
+    rp-->>-Browser: response
+```
+
+**Requirement**
```

Comment:
Instead of using bold text. Use markdown headers. This is valid for all the lines you use bold text that is actually a header.

Reply:
Updated.